frontend/src/features/commits/DatasetCommits.tsx

```diff
@@ -30,14 +30,25 @@ const DatasetCommits: React.FC<DatasetCommitsProps> = ({
   }, [dispatch, qriRef.username, qriRef.name])
 
   return (
-    <div className='my-1 mx-2 w-1/5 max-w-xs overflow-y-hidden flex flex-col text-xs'>
-      <header className='p-2 flex-grow-0'>
-        <Icon className='float-left mr-2 mt-1' icon='clock' />
-        <h3 className='text-lg font-bold'>History</h3>
+    <div className='pt-4 pl-7 overflow-y-hidden flex flex-col text-xs flex-shrink-0' style={{ width: '325px'}}>
+      <header className='flex-grow-0 mb-4'>
+        <h3 className='text-2xl text-qrinavy-500 font-black'>History</h3>
+        <div className='text-xs text-gray-400 tracking-wider'>{commits.length} versions</div>
       </header>
-      <ul className='block flex-grow overflow-y-auto pl-4 pr-2 pb-40'>
+      <ul className='block flex-grow overflow-y-auto pb-40 pr-8'>
+        <HistorySearchBox />
         {editable && <li><NewVersionButton qriRef={qriRef} /></li>}
-        {commits.map((logItem, i) => <DatasetCommitItem key={i} logItem={logItem} active={logItem.path === path} />)}
+        {commits.map((logItem, i) => {
+          return (
+            <DatasetCommitItem
+              key={i}
+              logItem={logItem}
+              active={logItem.path === path}
+              first={i === 0 && !editable}
+              last={i === (commits.length - 1)}
+            />
+          )
+        })}
```

Comment:
drop unnecessary return:

```suggestion
        {commits.map((logItem, i) => (
            <DatasetCommitItem
              key={i}
              logItem={logItem}
              active={logItem.path === path}
              first={i === 0 && !editable}
              last={i === (commits.length - 1)}
            />)}
```